main_test.go

```diff
@@ -21,10 +22,39 @@ func TestRootCmd(t *testing.T) {
 		t.Errorf("Execute failed: %v", err)
 	}
 
-	want, _ := os.ReadFile("fluentbit/fluentbit.test.conf")
+	fluentbitConfig, _ := os.ReadFile("fluentbit/test-configs/fluentbit-cloud-init-only.test.conf")
+	want := fmt.Sprintf("Main config:\n%s\nApplication config:%s", string(fluentbitConfig), "")
 
-	if got.String() != string(want) {
-		t.Error(diff.Diff(string(want), got.String()))
+	if got.String() != want {
+		t.Error(diff.Diff(want, got.String()))
+	}
+}
+
+func TestGenerateConfigs(t *testing.T) {
+
+	cloudInitOnly, _ := os.ReadFile("fluentbit/test-configs/fluentbit-cloud-init-only.test.conf")
+	cloudInitWithInclude, _ := os.ReadFile("fluentbit/test-configs/fluentbit-cloud-init-and-app.test.conf")
+	application, _ := os.ReadFile("fluentbit/test-configs/application-logs.test.conf")
+
+	withoutApplicationLogs := FluentbitConfig { MainConfigFile: string(cloudInitOnly), ApplicationConfigFile: "" }
+    withApplicationLogs := FluentbitConfig { MainConfigFile: string(cloudInitWithInclude), ApplicationConfigFile: string(application) }
+
+	var tests = []struct {
```

Comment:
I got the impression (from Google!) that using this 'table test' style was the recommended way to test multiple cases against the same function in `go` - let me know if there's a better way to do this!

Reply:
Yep definitely! https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go is a good description here.